core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -977,6 +981,55 @@ class CivilizationInfo {
         }
     }
 
+    fun getDiplomaticMarriageCost(): Int {
+        // https://github.com/Gedemon/Civ5-DLL/blob/master/CvGameCoreDLL_Expansion1/CvMinorCivAI.cpp, line 7812
+        var cost = when (gameInfo.gameParameters.gameSpeed) {
+            GameSpeed.Quick -> 335
+            GameSpeed.Standard -> 500
+            GameSpeed.Epic -> 750
+            GameSpeed.Marathon -> 1500
+        }
+        // Plus disband value of all units
+        for (unit in units) {
+            cost += unit.baseUnit.getDisbandGold(this)
+        }
+        // Round to lower multiple of 5
+        cost /= 5
+        cost *= 5
+
+        return cost
+    }
+
+    fun canBeMarriedBy(otherCiv: CivilizationInfo): Boolean {
+        return (!isDefeated()
+                && isCityState()
+                && getDiplomacyManager(otherCiv).relationshipLevel() == RelationshipLevel.Ally
+                && !otherCiv.getDiplomacyManager(this).hasFlag(DiplomacyFlags.MarriageCooldown)
+                && otherCiv.getMatchingUniques("Can spend Gold to annex or puppet a City-State that has been your ally for [] turns.").any()
+                && otherCiv.gold >= getDiplomaticMarriageCost())
+
+    }
+
+    fun diplomaticMarriage(otherCiv: CivilizationInfo) {
+        if (!canBeMarriedBy(otherCiv))  // Just in case
+            return
+
+        otherCiv.gold -= getDiplomaticMarriageCost()
+        otherCiv.addNotification("We have married into the ruling family of [${civName}], bringing them under our control.",
+            getCapital().location, civName, NotificationIcon.Diplomacy, otherCiv.civName)
+        for (civ in gameInfo.civilizations.filter { it != otherCiv })
+            civ.addNotification("[${otherCiv.civName}] has married into the ruling family of [${civName}], bringing them under their control.",
+                getCapital().location, civName, NotificationIcon.Diplomacy, otherCiv.civName)
+        for (unit in units)
+            unit.gift(otherCiv)
+        for (city in cities) {
+            city.moveToCiv(otherCiv)
+            city.isPuppet = true // Human players get a popup that allows them to annex instead
+            city.foundingCiv = "" // This is no longer a city-state
```

Comment:
Wouldn't we then want to set it to the nation that it has been married into?

Reply:
No, Austria should need to build courthouses.